packages/react-fresh/src/ReactFreshBabelPlugin.js

```diff
@@ -32,52 +32,173 @@ export default function(babel) {
     return typeof name === 'string' && name[0] >= 'A' && name[0] <= 'Z';
   }
 
-  function isComponentish(node) {
+  function findInnerComponents(inferredName, path, callback) {
+    const node = path.node;
     switch (node.type) {
-      case 'FunctionDeclaration':
-        return node.id !== null && isComponentishName(node.id.name);
-      case 'VariableDeclarator':
-        return (
-          isComponentishName(node.id.name) &&
-          node.init !== null &&
-          (node.init.type === 'FunctionExpression' ||
-            (node.init.type === 'ArrowFunctionExpression' &&
-              node.init.body.type !== 'ArrowFunctionExpression'))
+      case 'FunctionDeclaration': {
+        const id = node.id;
+        if (id === null) {
+          return false;
```

Comment:
This excludes default export functions that are declared like `export default function() {}`. Is this intended?

Reply:
Yeah. Without a name we don’t know if it’s likely a component or just some random utility.

Maybe we could support that. But then, you probably should name your components anyway? So they show up in stacks correctly. I know there are Babel plugins that do this by filename but honestly is it too much to ask that people give names to their functions? 🙂

Reply:
I think it's more than just `export default function() {}`. It also excludes _named_ functions exported with HOCs around them.

For example, this would match:
```js
export default function Foo() {
  return null;
}
```
And this would match:
```js
const Foo = function Foo(props, ref) {
  return null;
}
export default forwardRef(Foo);
```
But this would not:
```js
export default forwardRef(function Foo(props, ref) {
  return null;
});
```

I think the latter is a pattern I've used from time to time! 😁

Reply:
I'll be working more on these heuristics. I agree they don't make complete sense now.

Maybe I'll lean towards over-registering and then use some other heuristic for whether to propagate an update.

Reply:
Pushed a fix